### DISCUSSION TITLE
IPadapter: multi IPAdapter and multi input

### DISCUSSION THREAD

**Discussion by thibaudart** [huggingface/diffusers#6544] [Ideas]
hi

since a while, i use on comfyui a workflow with multi ipadapter (mainly one for face and one for style with different ipadapter model, different weights and different input image). 
moreover for the style one, i use a folder with 5 to 25 images. 
Results achieved with that ensure a very coherent style (like a lora) and very great consistency for character (face). 
replicating this process in pure diffusers could be very useful.

**Comment by asomoza** [comment]
there's an issue for it (#6318), you can follow the progress. I know you posted there but is to link both conversations.

**Comment by yiyixuxu** [comment]
Hi @thibaudart 

>  the style one, i use a folder with 5 to 25 images.

For multi-inputs , do you mean that you would need to pass multiple inputs to the same IP adapter and split the weights equality?  

e.g. would below API work? 

```python
pipeline.load_ip_adapter(["h94/IP-Adapter"], subfolder=["models"], weight_name=["ip-adapter_sd15.bin", "ip-adapter-full-face_sd15.bin"])
pipeline.set_ip_adapter_scale([0.3, 0.7])

images = pipeline(
    ...
    ip_adapter_image=[[style_img0, stype_img1], face_img],
    ...,
).images
```

it will be same as 

```python
pipeline.load_ip_adapter(["h94/IP-Adapter"], subfolder=["models"], weight_name=["ip-adapter_sd15.bin", "ip-adapter_sd15.bin", "ip-adapter-full-face_sd15.bin"])
pipeline.set_ip_adapter_scale([0.15, 0.15, 0.7])

images = pipeline(
    ...
    ip_adapter_image=[style_img0, stype_img1, face_img],
    ...,
).images
```

**Comment by thibaudart** [comment]
yes seems good!



T

**Comment by sayakpaul** [comment]
Can we also try marking the answers if that's the case? :)

**Comment by Nomination-NRB** [comment]
> hi
> 
> since a while, i use on comfyui a workflow with multi ipadapter (mainly one for face and one for style with different ipadapter model, different weights and different input image). moreover for the style one, i use a folder with 5 to 25 images. Results achieved with that ensure a very coherent style (like a lora) and very great consistency for character (face). replicating this process in pure diffusers could be very useful.

Hi, I am also interested in multiple IPAdapter control, can you share your comfyui workflow
